apps/zk-explorer/components/SidebarFilters/SidebarFilters.tsx

```diff
@@ -0,0 +1,206 @@
+import { FC, useEffect, useState } from 'react';
+import { Typography, CheckBox, Chip } from '@webb-tools/webb-ui-components';
+import { PropsOf } from '@webb-tools/webb-ui-components/types';
+import { cloneDeep } from 'lodash';
+import {
+  FilterCategory,
+  FilterCategoryItem,
+  FilterConstraints,
+  FilterOptionItem,
+} from './types';
+
+export type SidebarFiltersProps = PropsOf<'div'> & {
+  onConstraintsChange: (constraints: FilterConstraints) => void;
+};
+
+export const SidebarFilters: FC<SidebarFiltersProps> = ({
+  onConstraintsChange,
+}) => {
+  const [constraints, setConstraints] = useState<FilterConstraints>({
+    [FilterCategory.ProofSystem]: [],
+    [FilterCategory.Categories]: [],
+    [FilterCategory.License]: [],
+    [FilterCategory.LanguageOrFramework]: [],
+  });
+
+  // TODO: These are only for testing purposes. Remove these once the actual data is available.
+  const debugProofSystemsOptions: FilterOptionItem[] = [
```

Comment:
you can add more `useMemo` here and for other fake data below

Reply:
The mock data is static and there aren't any complex operations being made. Should I still proceed to use `useMemo` here?

Reply:
Yes we can leave this for now and make updates later